src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimSpecCallbackHandler.java

```diff
@@ -272,9 +274,14 @@ private CallbackResponse validateClaimant2Details(CallbackParams callbackParams)
     private CallbackResponse validateClaimantDetails(CallbackParams callbackParams,
                                                      Function<CaseData, Party> getApplicant) {
         CaseData caseData = callbackParams.getCaseData();
-        CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder();
         Party applicant = getApplicant.apply(caseData);
         List<String> errors = dateOfBirthValidator.validate(applicant);
+        if (applicant.getPrimaryAddress() != null) {
```

Comment:
We are modifying existing code, kindly put your changes behing the alunch darkly flag